compiler/src/dotty/tools/dotc/semanticdb/TypeOps.scala

```diff
@@ -0,0 +1,452 @@
+          // opaque type alias will be stored into the refinement of
+          // the self type of the enclosing class.
+          // Key: the tuple of
+          //   - self-type of enclosing class
+          //   - name of the opaque type
+          // Value: the symbol of the opaque type
+          // See: SymDenotation.opaqueToBounds
+          case cls: ClassInfo if sym.is(Flags.Opaque) =>
+            cls.classSymbol.asClass.givenSelfType match
+              case rt: RefinedType =>
+                refinementSymtab((rt, sym.name)) = sym
+              case _ => ()
+
+          case cls: ClassInfo if (cls.cls.name == tpnme.REFINE_CLASS) =>
+            enterRefined(sym.owner.owner.info)
+
+          // type x = Person { refinement }
+          case tb: TypeBounds =>
+            // tb = TypeBounds(
+            //   lo = RefinedType(...)
+            //   hi = RefinedType(...)
+            // )
+            enterRefined(tb.lo)
+            enterRefined(tb.hi)
+
+          // def s(x: Int): { refinement } = ...
+          case expr: ExprType =>
+            enterRefined(expr.resType)
+          case m: LambdaType =>
+            enterRefined(m.resType)
+          case _ => ()
+        }
+      if sym.exists && sym.owner.exists then
+        enterParamRef(sym.owner.info)
+        enterRefined(sym.owner.info)
+
+      def loop(tpe: Type): s.Signature = tpe match {
+        case mp: MethodOrPoly =>
+          def flatten(
+            t: Type,
+            paramss: List[List[SemanticSymbol]],
+            tparams: List[SemanticSymbol]
+          ): (Type, List[List[SemanticSymbol]], List[SemanticSymbol]) = t match {
+            case mt: MethodType =>
+              val syms: List[SemanticSymbol] = mt.paramNames.zip(mt.paramInfos).map { (name, info) =>
+                paramRefSymtab.lookup(mt, name).getOrElse {
+                  val fakeSym = TermParamRefSymbol(sym, name, info)
+                  registerFakeSymbol(fakeSym)
+                  fakeSym
+                }
+              }
+              flatten(mt.resType, paramss :+ syms, tparams)
+            case pt: PolyType =>
+              val syms: List[SemanticSymbol] = pt.paramNames.zip(pt.paramInfos).map { (name, info) =>
+                paramRefSymtab.lookup(pt, name).getOrElse {
+                  val fakeSym = TypeParamRefSymbol(sym, name, info)
+                  registerFakeSymbol(fakeSym)
+                  fakeSym
+                }
+              }
+              flatten(pt.resType, paramss, tparams ++ syms)
+            case other =>
+              (other, paramss, tparams)
+          }
+          val (resType, paramss, tparams) = flatten(mp, Nil, Nil)
+
+          val sparamss = paramss.map(_.sscope)
+          val stparams = tparams.sscopeOpt
+          s.MethodSignature(
+            stparams,
+            sparamss,
+            resType.toSemanticType(sym)
+          )
+
+        case cls: ClassInfo =>
+          val stparams = cls.cls.typeParams.sscopeOpt
```

Comment:
I was talking to @smarter and you may be able to resolve the `TypeTree` issue here by reusing the trees for the primary constructor type parameters (accessed through `sym.defTree`) - being careful to substitute symbols - maybe the same for exported type aliases (look in tree of rhs to get a type, but then exports of exports may pick up `TypeTree`)

Reply:
Thanks! I'll take a look at it in another issue / PR.
While the lack of symbol problem in TypeTree has been well resolved in terms of signature, I believe the lack of symbol occurrence in TypeTree is an issue to fix (it leads to the problem in rename feature in metals, and this kind of problem 
https://github.com/tanishiking/scalafix-unused/issues/5 in scalafix semantic rules).